src/server/req_holdjob.c

```diff
@@ -300,9 +300,54 @@ req_releasejob(struct batch_request *preq)
 		pjob->ji_modified = 1;	/* indicates attributes changed    */
 		svr_evaljobstate(pjob, &newstate, &newsub, 0);
 		(void)svr_setjobstate(pjob, newstate, newsub); /* saves job */
+
+	}
+
+	if ((jt == IS_ARRAY_ArrayJob) && (pjob->ji_ajtrk)) {
+		int i;
+		for(i = 0 ; i < pjob->ji_ajtrk->tkm_ct ; i++) {
+			job *psubjob = pjob->ji_ajtrk->tkm_tbl[i].trk_psubjob;
+			if (psubjob && (psubjob->ji_qs.ji_state == JOB_STATE_HELD)) {
+#ifndef NAS
+				old_hold = psubjob->ji_wattr[(int)JOB_ATR_hold].at_val.at_long;
+				rc =
+#endif
+					job_attr_def[(int)JOB_ATR_hold].
+					at_set(&psubjob->ji_wattr[(int)JOB_ATR_hold],
+					&temphold, DECR);
+#ifndef NAS /* localmod 105 Always reset etime on release */
+				if (!rc && (old_hold != psubjob->ji_wattr[(int)JOB_ATR_hold].at_val.at_long)) {
+#endif /* localmod 105 */
+#ifdef NAS /* localmod 105 */
+				{
+					attribute *etime = &psubjob->ji_wattr[(int)JOB_ATR_etime];
+					etime->at_val.at_long = time_now;
+					etime->at_flags |= ATR_VFLAG_SET|ATR_VFLAG_MODCACHE;
+#endif /* localmod 105 */
+					psubjob->ji_modified = 1;	/* indicates attributes changed    */
+					svr_evaljobstate(psubjob, &newstate, &newsub, 0);
+					(void)svr_setjobstate(psubjob, newstate, newsub); /* saves job */
+				}
+				if (psubjob->ji_wattr[(int)JOB_ATR_hold].at_val.at_long == 0)
+					job_attr_def[(int)JOB_ATR_Comment].at_free(&psubjob->ji_wattr[(int)JOB_ATR_Comment]);
+				(void)sprintf(log_buffer, msg_jobholdrel, pset, preq->rq_user,
+					preq->rq_host);
+				log_event(PBSEVENT_JOB, PBS_EVENTCLASS_JOB, LOG_INFO,
+					psubjob->ji_qs.ji_jobid, log_buffer);
+			}
+		}
+	}
+	if (pjob->ji_wattr[(int)JOB_ATR_hold].at_val.at_long == 0) {
```

Comment:
Instead of comparing it to 0 use HOLD_n

Reply:
Adressed in https://github.com/PBSPro/pbspro/pull/1119/commits/2d3a5a6fe3c51e0f88bb63f19c7624fbca4a48d0